gmsh_interop/reader.py

```diff
@@ -485,50 +537,57 @@ class GmshFileFormatError(RuntimeError):
     pass
 
 
-def read_gmsh(receiver, filename, force_dimension=None):
+def read_gmsh(
+        receiver: GmshMeshReceiverBase,
+        filename: str,
+        force_dimension: int | None = None) -> None:
     """Read a gmsh mesh file from *filename* and feed it to *receiver*.
 
     :param receiver: Implements the :class:`GmshMeshReceiverBase` interface.
     :param force_dimension: if not None, truncate point coordinates to
         this many dimensions.
     """
-    mesh_file = open(filename)
-    try:
-        result = parse_gmsh(receiver, mesh_file, force_dimension=force_dimension)
-    finally:
-        mesh_file.close()
-
-    return result
-
-
-def generate_gmsh(receiver, source, dimensions=None, order=None, other_options=(),
-            extension="geo", gmsh_executable="gmsh", force_dimension=None,
-            target_unit=None, output_file_name=None, save_tmp_files_in=None):
+    with open(filename) as mesh_file:
+        parse_gmsh(receiver, mesh_file, force_dimension=force_dimension)
+
+
+def generate_gmsh(
+        receiver: GmshMeshReceiverBase,
+        source: str | ScriptSource | FileSource | ScriptWithFilesSource,
+        dimensions: Literal[1, 2, 3] | None = None,
+        order: int | None = None,
+        other_options: tuple[str, ...] = (),
+        extension: str = "geo",
+        gmsh_executable: str = "gmsh",
+        force_dimension: int | None = None,
+        target_unit: Literal["M", "MM"] | None = None,
+        output_file_name: str | None = None,
+        save_tmp_files_in: str | None = None) -> None:
     """Run gmsh and feed the output to *receiver*.
 
     :arg receiver: a class that implements the :class:`GmshMeshReceiverBase`
         interface.
     :arg source: an instance of :class:`ScriptSource` or :class:`FileSource`.
     """
     from gmsh_interop.runner import GmshRunner
+
     runner = GmshRunner(source, dimensions, order=order,
             other_options=other_options, extension=extension,
             gmsh_executable=gmsh_executable,
             target_unit=target_unit,
             output_file_name=output_file_name,
             save_tmp_files_in=save_tmp_files_in)
 
-    runner.__enter__()
-    try:
-        result = parse_gmsh(receiver, runner.output_file,
-                force_dimension=force_dimension)
-    finally:
-        runner.__exit__(None, None, None)
+    with runner:
+        parse_gmsh(
+            receiver,
+            runner.output_file,
+            force_dimension=force_dimension)
```

Comment:
Is this equivalent? I'm not quite sure what the original code tried to do..

Reply:
I think the original code tried to stay compatible with pre-context-manager Pythons.